src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3824,6 +3745,75 @@ private void TrackNullableStateOfNullableValue(BoundExpression node, BoundExpres
             }
         }
 
+        private void TrackNullableStateOfTupleConversion(BoundConversion node)
+        {
+            Debug.Assert(node.ConversionKind == ConversionKind.ImplicitTuple || node.ConversionKind == ConversionKind.ExplicitTuple);
+
+            var operand = node.Operand;
+            var valueTuple = operand.Type as TupleTypeSymbol;
+            if (valueTuple is null)
+            {
+                return;
+            }
+
+            int valueSlot = MakeSlot(operand);
+            if (valueSlot < 0)
+            {
+                return;
+            }
+
+            int slot = GetOrCreateObjectCreationPlaceholderSlot(node);
+            if (slot < 0)
+            {
+                return;
+            }
+
+            var conversions = node.Conversion.UnderlyingConversions;
+            var targetElements = ((TupleTypeSymbol)node.Type).TupleElements;
+            var valueElements = valueTuple.TupleElements;
+            int n = valueElements.Length;
+            for (int i = 0; i < n; i++)
+            {
+                trackConvertedValue(targetElements[i], conversions[i], valueElements[i]);
+            }
+
+            void trackConvertedValue(FieldSymbol targetField, Conversion conversion, FieldSymbol valueField)
```

Comment:
>trackConvertedValue [](start = 17, length = 19)

How does this function relate to `ApplyConversion(..., trackNullability: true)`? Could we just leverage `ApplyConversion`?

Reply:
This function should probably be merged with `ApplyConversion`. I'm attempting to do that with the next PR.

---
In reply to: [267979386](https://github.com/dotnet/roslyn/pull/34303#discussion_r267979386) [](ancestors = 267979386)